src/Analyzers/CSharp/Tests/UseIndexOrRangeOperator/UseRangeOperatorTests.cs

```diff
@@ -842,5 +842,108 @@ void Goo(string s, int[] bar)
                 FixedCode = fixedSource,
             }.RunAsync();
         }
+
+        [WorkItem(47183, "https://github.com/dotnet/roslyn/issues/47183")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseRangeOperator)]
+        public async Task TestExpressionWithNullConditionalAccess()
+        {
+            var source =
+@"
+#nullable enable
+public class Test
+{
+    public string? M(string? arg)
+        => arg?.Substring([|42|]);
+}";
+            var fixedSource =
+@"
+#nullable enable
+public class Test
+{
+    public string? M(string? arg)
+        => arg?[42..];
+}";
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
+                TestCode = source,
+                FixedCode = fixedSource,
+            }.RunAsync();
+        }
+
+        [WorkItem(47183, "https://github.com/dotnet/roslyn/issues/47183")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseRangeOperator)]
+        public async Task TestExpressionWithNullConditionalAccessWithPropertyAccess()
+        {
+            var source =
+@"
+public class Test
+{
+    public int M(string arg)
+        => arg.Substring([|42|]).Length;
+}";
+            var fixedSource =
+@"
+public class Test
+{
+    public int M(string arg)
+        => arg[42..].Length;
+}";
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
+                TestCode = source,
+                FixedCode = fixedSource,
+            }.RunAsync();
+        }
+
+        [WorkItem(47183, "https://github.com/dotnet/roslyn/issues/47183")]
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsUseRangeOperator)]
+        [InlineData(
+            "c.Prop.Substring([|42|])",
+            "c.Prop[42..]")]
+        [InlineData(
+            "c.Prop.Substring([|42, c.Prop.Length - 1|])",
+            "c.Prop[42..^(-41)]")]
```

Comment:
This one looks odd. I this right or another bug?

Reply:
I think it's right given that substring is start/length, not start/end. 

Test with likely be clearer as: .Substring(1, c.Prop.Length-2)

I.e. an attempt to trim the first and last char from the string.

Reply:
The range looks odd because the SubString call is bogus (OutOfRangeException). Must be more like `c.Prop.Substring(42, c.Prop.Length - 43)`. I will change the test case to more sensible values.

Reply:
I change the values in 22abd62b